src/components/Account/AccountProfileContainer.tsx

```diff
@@ -0,0 +1,255 @@
+    accountAddress: ownProps.match.params.accountAddress,
+    accountInfo: state.arc.accounts[state.web3.ethAccountAddress + "-" + queryValues.daoAvatarAddress],
+    accountProfile: state.profiles[state.web3.ethAccountAddress],
+    currentAccountAddress: state.web3.ethAccountAddress,
+    dao: queryValues.daoAvatarAddress ? state.arc.daos[queryValues.daoAvatarAddress as string] : null
+  };
+};
+
+interface IDispatchProps {
+  showNotification: typeof showNotification;
+  updateProfile: typeof profileActions.updateProfile;
+}
+
+const mapDispatchToProps = {
+  showNotification,
+  updateProfile: profileActions.updateProfile
+};
+
+type IProps = IStateProps & IDispatchProps;
+
+interface IState {
+  genCount: number;
+  ethCount: number;
+}
+
+interface FormValues {
+  description: string;
+  githubURL: string;
+  name: string;
+}
+
+class AccountProfileContainer extends React.Component<IProps, IState> {
+
+  constructor(props: IProps) {
+    super(props);
+
+    this.state = {
+      ethCount: null,
+      genCount: null
+    };
+  }
+
+  public async componentWillMount() {
+    const { accountAddress, dao } = this.props;
+    const web3 = await Arc.Utils.getWeb3();
+    const getBalance = promisify(web3.eth.getBalance);
+    const ethBalance = await getBalance(accountAddress);
+
+    let votingMachineInstance: Arc.GenesisProtocolWrapper;
+    if (dao) {
+      const contributionRewardInstance = await Arc.ContributionRewardFactory.deployed();
+      const votingMachineAddress = (await contributionRewardInstance.getSchemeParameters(dao.avatarAddress)).votingMachineAddress;
+      votingMachineInstance = await Arc.GenesisProtocolFactory.at(votingMachineAddress);
+    } else {
+      votingMachineInstance = await Arc.GenesisProtocolFactory.deployed();
+    }
+    const stakingTokenAddress = await votingMachineInstance.contract.stakingToken();
+    const stakingToken = await (await Arc.Utils.requireContract("StandardToken")).at(stakingTokenAddress) as any;
+    const genBalance = await stakingToken.balanceOf(accountAddress);
+
+    this.setState({ ethCount: Util.fromWei(ethBalance), genCount: Util.fromWei(genBalance)});
+  }
+
+  public copyAddress = () => {
+    const { showNotification, accountAddress } = this.props;
+    Util.copyToClipboard(accountAddress);
+    showNotification(NotificationStatus.Success, `Copied to clipboard!`);
+  }
+
+  public async handleSubmit(values: FormValues, { props, setSubmitting, setErrors }: any ) {
+    const { accountAddress, updateProfile } = this.props;
+
+    const web3 = await Arc.Utils.getWeb3();
+    const text = "Please sign in to Alchemy";
+    const msg = ethUtil.bufferToHex(Buffer.from(text, 'utf8'));
+    const fromAddress = this.props.accountAddress;
+
+    let signature = localStorage.getItem("signature-" + fromAddress);
+    if (!signature) {
+      const method = 'personal_sign';
+      const sendAsync = promisify(web3.currentProvider.sendAsync);
+      const params = [msg, fromAddress];
```

Comment:
you should sign *all the parameters you send*, otherwise it provides very little security.